core/src/main/scala/kafka/server/DelayedRemoteFetch.scala

```diff
@@ -79,7 +80,12 @@ class DelayedRemoteFetch(remoteFetchTask: Future[Void],
   override def onExpiration(): Unit = {
     // cancel the remote storage read task, if it has not been executed yet
     val cancelled = remoteFetchTask.cancel(true)
-    if (!cancelled) debug(s"Remote fetch task for for RemoteStorageFetchInfo: $remoteFetchInfo could not be cancelled and its isDone value is ${remoteFetchTask.isDone}")
+    if (!cancelled) debug(s"Remote fetch task for RemoteStorageFetchInfo: $remoteFetchInfo could not be cancelled and its isDone value is ${remoteFetchTask.isDone}")
+
+    if (fetchParams.isFromFollower)
```

Comment:
I understand it is a defensive check and this instance is never created if it is a follower fetch. If we want a defensive check, it should be created when `DelayedRemoteFetch` is created instead of creating it in this callback.

Reply:
Agree! Let me fix that in my follow-up PR.